src/Kiota.Builder/Refiners/CommonLanguageRefiner.cs

```diff
@@ -224,6 +225,14 @@ protected static void AddGetterAndSetterMethods(CodeElement current, HashSet<Cod
             var propertyOriginalName = (currentProperty.IsNameEscaped ? currentProperty.SerializationName : current.Name)
                                         .ToFirstCharacterLowerCase();
             var accessorName = refineAccessorName(propertyOriginalName.CleanupSymbolName().ToFirstCharacterUpperCase());
+            if (!supportsOverloading)
+            {
+                var existing = parentClass.FindChildByName<CodeProperty>(accessorName);
```

Comment:
this is not going to be deterministic depending on which one gets added first. We should have a routine here that normalizes both properties.

Reply:
In this case its checking the model property and not the method names, i.e getters and setters are generated after all the properties already exist for eaxample in the unit test

Reply:
Thanks for clarifying.